data-prepper-core/src/main/java/com/amazon/dataprepper/peerforwarder/PeerForwarderAppConfig.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.peerforwarder;
+
+import com.amazon.dataprepper.parser.model.DataPrepperConfiguration;
+import org.springframework.context.annotation.Bean;
+
+public class PeerForwarderAppConfig {
+
+    @Bean
+    public PeerForwarderConfiguration peerForwarderConfiguration(final DataPrepperConfiguration dataPrepperConfiguration) {
```

Comment:
I believe that you need this annotation on `dataPrepperConfiguration`: `@Autowired(required = false)`. Without this, Spring will fail if it is null.